### DISCUSSION TITLE
I did found some French in a filter instead to be in English

### DISCUSSION THREAD

**Discussion by PatLabGit** [lbalazscs/Pixelitor#318] [General]
Hello, 

In the Filter > Render > Geometry > Truchet Tiles... Patterns' names are French numbers' naming (one to twenty one), I speak French, thus that was kind of funny to see it, but maybe it's better to translate them in proper English for people to understand the number of pattern?

![screenshot_20230915-192728](https://github.com/lbalazscs/Pixelitor/assets/47497693/b9fa7548-e6ca-493a-b85e-c9889abe8126)

Last but not least, your made a wonderful piece of software, thank you!

**Comment by Minecraftian14** [comment]
How about naming them with elements like Hydrogen, Helium, Lithium...

Or maybe Amethyst, Baryte, Calcite, Diamond, Epidote...

**Reply by Ayushh1023** [reply]
Like this?...👀 
![Precious_Presets](https://github.com/lbalazscs/Pixelitor/assets/99321679/895926cc-6ee8-43bd-8cfa-dcc6ff39368e)

**Reply by Minecraftian14** [reply]
Yeah, but more like... In alphabetical order.
A - Amethyst
B - Baryte
C - Citrine
D - Diamond
...

**Comment by lbalazscs** [comment]
The French names were intentional, but they weren't necessarily a good idea. When I was implementing it, I thought it would be funny to give French names to those settings, given that Truchet was French. The pattern names are entirely made-up by me; I have no idea of how Truchet named the patterns. A week later, those names didn't seem as funny anymore, but I couldn't come up with a better idea for naming them, so I left them as they were.

I like @Minecraftian14's idea of using gemstone names, and I also agree that they should be in alphabetical order, but it's not necessary for each letter to have a corresponding name - the numbers don't mean anything. I used numbers only because my French is very basic...

I see that you've figured out how to change the code, so you might as well go ahead and send a pull request. 

One downside of changing the names is that it will break the compatibility of filter presets and smart filters. However, there's a place in the code where we can put automatic migration steps; this has been necessary a few times already. The place is in UserPreset.java, in the "get(String key)" method, in the if (value == null) block. if (value == null) means that the key (the new name) didn't work (because the preset or the smart filter was saved in a previous version, with the old name), so we try the old name as well.

**Reply by lbalazscs** [reply]
Update: In the meantime, I have realized that the migration trick described above wouldn't work in this case because the "value" is changing, not the "key." (Until now, I have always changed the name of a property.) Automatic migration would be more complicated, and perhaps it's best to abandon the idea of migrating. It's unlikely that many presets are saved, and smart filters are an experimental feature; they might lose their compatibility anyway.

**Comment by lbalazscs** [comment]
Oh, I see only now that while the original comment was made by @PatLabGit, the screenshot was sent by @Ayushh1023. Well, if anyone feels like sending a pull request, it's fine by me.

**Comment by PatLabGit** [comment]
Is it possible with the triangles patterns in Truchet to get "Hounds tooth - pied de poule" pattern?


![screenshot_20230918-104130](https://github.com/lbalazscs/Pixelitor/assets/47497693/d5ffcfd6-40ab-47d9-8c25-6dd95d13540b)

**Comment by Minecraftian14** [comment]
Everthing is possible... This specific pattern requires a little more than tweaking the code.
A simple workaround is adding two new plain tiles - a blank and a filled one.
Here's what I got:

![image](https://github.com/lbalazscs/Pixelitor/assets/52451860/d987f1c3-8818-433d-9f8a-45dcbaab70ed)

![image](https://github.com/lbalazscs/Pixelitor/assets/52451860/8873215a-42db-45bf-961a-5458557aa427)

**Reply by lbalazscs** [reply]
Very interesting, even if, strictly speaking, it's not Truchet tiling, since the new tiles aren't created by rotating an initial tile. I wonder if this could be added as a feature to the filter anyway, but I'm not sure if the idea can be generalized to the other pattern types ("quarter circles" and "diagonals")

**Reply by Minecraftian14** [reply]
That was exactly the problem!! Yes, it is certainly nowhere pleasant to look at:

<img width="45%" src="https://github.com/lbalazscs/Pixelitor/assets/52451860/aede60c6-48ad-4393-8fb1-c7490c6d0c8d">
<img width="45%" src="https://github.com/lbalazscs/Pixelitor/assets/52451860/96118b7b-5688-42e4-bbe7-c71bb55ced61">

However, I tried on some work arounds, like:

<img width="45%" src="https://github.com/lbalazscs/Pixelitor/assets/52451860/21234cd9-d608-4dce-ae88-f6af0ee100b9">
<img width="45%" src="https://github.com/lbalazscs/Pixelitor/assets/52451860/4873ed40-4257-4bf7-8bd2-a478c938b20d">

Should I add a PR?

**Reply by lbalazscs** [reply]
Yes, the last two images look cool to me!

**Comment by LinuxBeaver** [comment]
I'd rather not break presets with this plugin but if it happens, oh well.

**Comment by PatLabGit** [comment]
> Yes, the last two images look cool to me!

For me as well, they are really cool, very nice! (I even like the second one)

**Reply by lbalazscs** [reply]
In the meantime, @Minecraftian14 has implemented it. If you can build Pixelitor from the source, then you can use the new patterns. He also intends to make further improvements, allowing the editing of the patterns instead of just selecting them from a list. So your initial comment has started a lot of changes!

(Some of the discussions happen in the PR comments, and others on Discord: https://discord.gg/SXaxYnBSTv)

**Reply by PatLabGit** [reply]
> So your initial comment has started a lot of changes!

Wow, I see, I did not expect that, but that's really cool.
I'm not good to compile, even I'm 100% using Linux for many years, I'm just a user, no worries I'll wait the next update ;)
For Discord, I can't, they are asking my phone number :( , do you have a reddit instead?

**Reply by lbalazscs** [reply]
There is a way to get the latest jar file without building it yourself, as described at https://github.com/lbalazscs/Pixelitor/wiki/Downloading-the-latest-development-build

Note that while the latest Pixelitor release requires Java 17+, the newest builds require Java 21 (which was released about a month ago)

I do use reddit, I guess I could create a subreddit for Pixelitor. But the Discord doesn't have a lot of traffic, probably a subreddit would be empty as well. So I suggest that you just reply here, or create a new discussion, if you have any questions or feedback.
